packages/devextreme/js/__internal/ui/hierarchical_collection/data_adapter.ts

```diff
@@ -31,49 +30,59 @@ interface LangParams {
   locale?: string;
   collator?: Intl.Collator;
 }
-export interface BaseDataAdapterOptions {
-  rootValue: ItemKey | undefined;
-  multipleSelection: boolean;
-  recursiveSelection?: boolean;
-  recursiveExpansion?: boolean;
-  searchValue: string;
-}
 
-export interface DataAdapterOptions extends BaseDataAdapterOptions {
+export interface DataAdapterOptions {
   dataAccessors: DataAccessors;
   items: ItemData[];
-  dataConverter: HierarchicalDataConverter;
-  onNodeChanged: (node: InternalNode) => void;
+
+  rootValue: ItemKey;
+  multipleSelection: boolean;
+  recursiveSelection: boolean;
+  recursiveExpansion: boolean;
+  searchValue: string;
+
   selectionRequired?: boolean;
-  dataType: DataType;
+  dataType: DataStructure;
   sort: SortOption | SortOption[] | null;
   langParams?: LangParams;
-  searchMode: SearchMode;
   searchExpr?: string | Function | (string | Function)[] | ((item: ItemData) => unknown);
+
+  dataConverter: HierarchicalDataConverter;
+  onNodeChanged: (node: InternalNode) => void;
+
+  searchMode: SearchMode;
 }
 
 SearchBoxController.setEditorClass(TextBox);
 
 class DataAdapter {
-  options: DataAdapterOptions;
-
-  _selectedNodesKeys: ItemKey[];
-
-  _expandedNodesKeys: ItemKey[];
-
-  _dataStructure: (InternalNode | null)[];
-
-  _initialDataStructure: InternalNode[] = [];
-
-  constructor(options: Partial<DataAdapterOptions>) {
-    this.options = {} as DataAdapterOptions;
-    extend(this.options, this._defaultOptions(), options);
+  options: DataAdapterOptions = {
+    dataAccessors: {} as DataAccessors,
+    items: [],
+    multipleSelection: true,
+    recursiveSelection: false,
+    recursiveExpansion: false,
+    rootValue: 0,
+    searchValue: '',
+    dataType: 'tree',
+    searchMode: 'contains',
+    dataConverter: new HierarchicalDataConverter(),
+    onNodeChanged: noop,
+    sort: null,
+  };
+
+  _selectedNodesKeys: ItemKey[] = [];
+
+  _expandedNodesKeys: ItemKey[] = [];
+
+  _dataStructure: (InternalNode | null)[] = [];
+
+  _initialDataStructure: (InternalNode | null)[] = [];
+
+  constructor(options: DataAdapterOptions) {
+    extend(this.options, options);
```

Comment:
Can we replace it with spread? we do not use deep copy here
```suggestion
    this.options = {
        ...this.options, 
        ...options,
    };
```